R/post_ts.R

```diff
@@ -9,37 +9,60 @@
 #' @param ... args passed to \code{\link[sbtools]{session_check_reauth}}
 #' @author Luke Winslow, Corinna Gries, Jordan S Read
 #' @import sbtools
+#' @importFrom lubridate tz
+#' @importFrom unitted u v get_units
 #' @examples
 #' \dontrun{
-#' files <- stage_nldas_ts(sites = c("nwis_06893820","nwis_01484680"), variable = "baro", 
+#' files <- stage_nldas_ts(sites = c("nwis_06893820","nwis_01484680"), var = "baro", 
 #'                         times = c('2014-01-01 00:00','2014-01-01 05:00'))
-#' post_ts(files, session = sbtools::authenticate_sb())
+#' post_ts(files)
 #' }
 #' @export
-post_ts = function(files, on_exists=c("stop", "skip", "replace"), verbose=TRUE, ...){
+post_ts = function(files, on_exists=c("stop", "skip", "replace", "merge"), verbose=TRUE, ...){
   
-  # check inputs
+  # check inputs & session
   if(is.null(files)) return(invisible(NULL))
   on_exists <- match.arg(on_exists)
+  session_check_reauth(...)
   
+  # loop through files, posting each
   posted_items <- sapply(1:length(files), function(i) {
-    if(verbose) message('posting file ', files[i])
+    if(verbose) message('preparing to post file ', files[i])
     
     # parse the file name to determine where to post the file
-    out <- parse_ts_path(files[i], out = c('ts_name','var_src','site_name','file_name'))
-    site <- out$site_name
+    out <- parse_ts_path(files[i], out = c('ts_name','var','src','var_src','site_name','file_name','dir_name'))
     
-    # Check if item already exists
-    ts_id <- locate_ts(var_src=out$var_src, site_name=out$site_name, ...)
+    # check & respond if item already exists
+    ts_id <- locate_ts(var_src=out$var_src, site_name=out$site_name)
     if (!is.na(ts_id)) {
       if(verbose) message('the ', out$ts_name, ' timeseries for site ', out$site_name, ' already exists')
-      switch(on_exists,
-             "stop"={ stop('item already exists and on_exists="stop"') },
-             "skip"={ next },
-             "append"={ stop("oops - append isn't implemented yet") },
-             "replace"={ if(verbose) message("deleting timeseries item before replacement")
-               delete_ts(out$var_src, out$site_name, verbose=verbose, ...)
-             })
+      switch(
+        on_exists,
+        "stop"={ stop('item already exists and on_exists="stop"') },
+        "skip"={ 
+          if(verbose) message("skipping timeseries item already on ScienceBase: ", ts_id)
+          return(NA) 
+        },
+        "replace"={ 
+          if(verbose) message("deleting timeseries item before replacement: ", ts_id)
+          delete_ts(out$var_src, out$site_name, verbose=verbose)
+        },
+        "merge"={ 
```

Comment:
yep. merge instead of append because of ted's question about being able to include data from earlier as well as later in time.